### PR TITLE
Ticket page design details

### LINKED ISSUE #449: ticket page design details
The following enhancements and fixes need to be made to the ticket page
* wallet title breaking into new line
The wallet title on revoked tickets breaks into a new line, causing the ticket card to have a different height than other tickets with different status
Screenshot 2021-05-31 at 16 32 41

* ticket progressbar
Currently, the ticket progress bar on a ticket card doesn't represent live data. It should display the lifecycle of a ticket according to its status. Same thing should be applied on the time displayed on the ticket. The clock icon should be displayed as well. 

Below is how the ticket cards should look like
Screenshot 2021-05-31 at 16 41 52
Screenshot 2021-05-31 at 16 42 06

### PR DESCRIPTION
**Pull request by tuhalang**
Resolve #449 
- Fix breaking into new line of wallet title in ticket card
- Add live data for progress bar and time
- Add icon timer
Before:
image
After:
image

### CODE REVIEW

**Comment by oshorefueled** [comment]
* The ticket status and wallet name on the ticket cards still break into a new line on my display. 
* If the lifecycle of a ticket is complete, it shouldn't display the time left till completion. Kindly check the mockups
   on Figma to correct the current implementation.
* The ticket should read 14d not 14 days. Hours should be 8h not < 1 day.
Screenshot 2021-06-12 at 21 29 52

File: ui/page/tickets/utils.go [hunk 1]
```diff
@@ -461,3 +511,38 @@ func createClickGestures(count int) []*gesture.Click {
 	}
 	return gestures
 }
+
+func getPercentConfirmation(l *load.Load, blockHeight int32) int {
+
```
Comment (Sirmorrison) [inline ui/page/tickets/utils.go]: Remove space

File: ui/decredmaterial/theme.go [hunk 2]
```diff
@@ -269,7 +269,7 @@ func Fill(gtx layout.Context, col color.NRGBA) layout.Dimensions {
 	return fill(gtx, col)
 }
 
-// mulAlpha scales all color components by alpha/255.
+// mulAlpha scales all color components.go by alpha/255.
```
Comment (Sirmorrison) [inline ui/decredmaterial/theme.go]: Kindly revert this... the  comments is talking about the component not the file.

File: ui/load/load.go [hunk 3]
```diff
@@ -1,5 +1,5 @@
-// The load package contains data structures that are shared by components in the ui package. It is not a dumping ground
-// for code you feel might be shared with other components in the future. Before adding code here, ask yourself, can
+// The load package contains data structures that are shared by components.go in the ui package. It is not a dumping ground
+// for code you feel might be shared with other components.go in the future. Before adding code here, ask yourself, can
```
Comment (Sirmorrison) [inline ui/load/load.go]: Kindly revert this... the comments is talking about the component not the file.

File: ui/page/components/components.go [hunk 4]
```diff
@@ -1,4 +1,4 @@
-// components contain layout code that are shared by multiple pages but aren't widely used enough to be defined as
+// components.go contain layout code that are shared by multiple pages but aren't widely used enough to be defined as
```
Comment (Sirmorrison) [inline ui/page/components/components.go]: Kindly revert this... the comments is talking about the component package not the file.

File: ui/page/components/components.go [hunk 5]
```diff
@@ -571,7 +571,7 @@ func toolTipContent(inset layout.Inset, body layout.Widget) layout.Widget {
 	})
 }*/
 
-// ticketActivityRow layouts out ticket info, display ticket activities on the tickets_page and tickets_activity_page
+// ticketActivityRow layouts out ticket info, display ticket activities on the tickets_page.go and tickets_activity_page
```
Comment (Sirmorrison) [inline ui/page/components/components.go]: Kindly revert this.

File: ui/page/tickets/utils.go [hunk 6]
```diff
@@ -209,20 +210,54 @@ func ticketCard(gtx layout.Context, l *load.Load, t *wallet.Ticket, tooltip tool
...
 								return wTimeLabel.Layout(gtx, func(gtx C) D {
+
+									blockHeight := t.Info.BlockHeight
+									totalTime := getTimeToDone(l, blockHeight)
+
+									if totalTime == 0 {
+										return layout.Inset{}.Layout(gtx, func(gtx C) D {
+											return layout.Dimensions{}
+										})
+									}
```
Comment (Sirmorrison) [inline ui/page/tickets/utils.go]:
```suggestion
									if totalTime == 0 {
										return layout.Dimensions{}
									}
```

File: ui/page/tickets/utils.go [hunk 7]
```diff
@@ -274,9 +316,11 @@ func ticketCard(gtx layout.Context, l *load.Load, t *wallet.Ticket, tooltip tool
 								})
 							}),
 							layout.Rigid(func(gtx C) D {
-								return layout.Flex{Alignment: layout.Middle}.Layout(gtx,
+								return layout.Flex{Alignment: layout.Middle, Spacing: layout.SpaceBetween}.Layout(gtx,
 									layout.Rigid(func(gtx C) D {
+
 										txt := l.Theme.Label(values.MarginPadding14, t.Info.Status)
+										txt.TextSize = unit.Sp(12)
```
Comment (Sirmorrison) [inline ui/page/tickets/utils.go]:
```suggestion
										txt.TextSize = values.TextSize12
```

File: ui/page/tickets/utils.go [hunk 8]
```diff
@@ -298,6 +342,7 @@ func ticketCard(gtx layout.Context, l *load.Load, t *wallet.Ticket, tooltip tool
 									layout.Rigid(func(gtx C) D {
 										txt := l.Theme.Label(values.MarginPadding14, t.WalletName)
 										txt.Color = l.Theme.Color.Gray
+										txt.TextSize = unit.Sp(14)
```
Comment (Sirmorrison) [inline ui/page/tickets/utils.go]:
```suggestion
										txt.TextSize = values.TextSize14
```

File: ui/page/tickets/utils.go [hunk 9]
```diff
@@ -360,7 +410,7 @@ func ticketCard(gtx layout.Context, l *load.Load, t *wallet.Ticket, tooltip tool
 	})
 }
 
-// ticketActivityRow layouts out ticket info, display ticket activities on the tickets_page and tickets_activity_page
+// ticketActivityRow layouts out ticket info, display ticket activities on the tickets_page.go and tickets_activity_page
```
Comment (Sirmorrison) [inline ui/page/tickets/utils.go]: Revert this

**Review by Sirmorrison — CHANGES_REQUESTED** [review]

File: ui/page/tickets/utils.go [hunk 10]
```diff
@@ -340,7 +383,12 @@ func ticketCard(gtx layout.Context, l *load.Load, t *wallet.Ticket, tooltip tool
 											})
 										}),
 										layout.Rigid(func(gtx C) D {
-											txt.Text = t.DaysBehind
+											timeBehind, unit := getTimeBehind(t.DateTime)
+											if timeBehind == 0 && unit == "h" {
+												return layout.Dimensions{}
+											}
+
+											txt.Text = fmt.Sprintf("%d%s", timeBehind, unit)
```
Comment (Sirmorrison) [inline ui/page/tickets/utils.go:391]: This does not seem to be taking effect as the ticket age still shows as `day` instead of `d`
 
image

**Review by Sirmorrison — CHANGES_REQUESTED** [review]
> * The ticket status and wallet name on the ticket cards still break into a new line on my display.
> * If the lifecycle of a ticket is complete, it shouldn't display the time left till completion. Kindly check the mockups
>   on Figma to correct the current implementation.
> * The ticket should read 14d not 14 days. Hours should be 8h not < 1 day.
> 
> Screenshot 2021-06-12 at 21 29 52

- These comments does not seem to have been resolved.
- The progress bar also does not seem to work from my end.

File: ui/page/tickets/utils.go (hunk 10 above)
Comment (tuhalang) [reply ui/page/tickets/utils.go:391]: image

Comment (tuhalang) [reply ui/page/tickets/utils.go:391]: i see every things is ok

Comment (Sirmorrison) [reply ui/page/tickets/utils.go:391]: kindly try purchasing a new ticket and you will see this
image

**Comment by Clariceco** [comment]
@tuhalang what is the status of this?

**Comment by Sirmorrison** [comment]
Most of the changes here have been addressed on PR #567. We will be closing this when #567 is merged.

**Pull request closed without merging**